examples/benchmark/models.py

```diff
@@ -0,0 +1,194 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+import abc
+import os
+import random
+import time
+
+import torch
+import torch.nn as nn
+import torchvision
+from PIL import Image
+from omegaconf import DictConfig
+from s3torchconnector import S3Reader, S3Checkpoint
+from torchvision.transforms import v2
+from transformers import ViTForImageClassification
+
+from benchmark_utils import ExperimentResult
+
+
+class ModelInterface(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def load_sample(self, sample):
+        """Transform given sample (a File-Like-Object) to a model-input"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def train(
+        self, dataloader: torch.utils.data.dataloader, epochs: int
+    ) -> ExperimentResult:
+        """Train the model using given dataloader for number of epochs"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def save(self):
+        """Save checkpoint"""
+        raise NotImplementedError
+
+
+# This is not really a training model as it does not train anything
```

Comment:
nit: We probably should make these comments python-style i.e. Below the class description, with a ''' block.